velox/common/memory/MappedMemory.h

```diff
@@ -397,19 +399,35 @@ class MappedMemory : public std::enable_shared_from_this<MappedMemory> {
 
   virtual void freeContiguous(ContiguousAllocation& allocation) = 0;
 
-  // Allocates 'bytes' contiguous bytes and returns the pointer to the first
-  // byte. If 'bytes' is less than 'kMaxMallocBytes', delegates the allocation
-  // to malloc. If the size is above that and below the largest size classes'
-  // size, allocates one element of the next size classes' size. If 'size' is
-  // greater than the largest size classes' size, calls allocateContiguous().
-  // Returns nullptr if there is no space. The amount to allocate is subject to
-  // the size limit of 'this'. This function is not virtual but calls the
-  // virtual functions allocate and allocateContiguous, which can track sizes
-  // and enforce caps etc.
-  virtual void* FOLLY_NULLABLE allocateBytes(uint64_t bytes);
-
-  // Frees memory allocated with allocateBytes().
-  virtual void freeBytes(void* FOLLY_NONNULL p, uint64_t size) noexcept;
+  /// Allocates 'bytes' contiguous bytes and returns the pointer to the first
+  /// byte. If 'bytes' is less than 'kMaxMallocBytes', delegates the allocation
+  /// to malloc. If the size is above that and below the largest size classes'
+  /// size, allocates one element of the next size classes' size. If 'size' is
+  /// greater than the largest size classes' size, calls allocateContiguous().
+  /// Returns nullptr if there is no space. The amount to allocate is subject to
+  /// the size limit of 'this'. This function is not virtual but calls the
+  /// virtual functions allocateNonContiguous and allocateContiguous, which can
+  /// track sizes and enforce caps etc.
+  ///
+  /// NOTE: 'alignment' must be power of two and in range of [0, kMaxAlignment].
+  virtual void* FOLLY_NULLABLE
+  allocateBytes(uint64_t bytes, uint16_t alignment = 0) = 0;
+
+  /// Allocates a zero-filled contiguous bytes.
+  virtual void* FOLLY_NULLABLE allocateZeroFilled(uint64_t bytes);
+
+  /// Allocates 'newSize' contiguous bytes. If 'p' is not null, this function
+  /// copies std::min(size, newSize) bytes from 'p' to the newly allocated
+  /// buffer and free 'p' after that.
+  virtual void* FOLLY_NULLABLE reallocateBytes(
+      void* FOLLY_NONNULL p,
+      int64_t size,
```

Comment:
Is 'size' aligned or not? How would the caller of allocateBytes with non-zero alignment would know how much bytes to free?

Reply:
'size' needs to be multiple of alignment and otherwise the allocation will fail. MappedMemory won't change the allocation size.